phsb5321/aca.so
# Feature/implementation of books services

## Feature/implementation of book services

This PR adds the implementation of the remaining book services to the project. The following changes were made:

- A `GetBookService` class in `app/services/books_services/get_book_service.py` that retrieves a book by ID from the database.
- A `ListBooksService` class in `app/services/books_services/list_books_service.py` that retrieves a list of all books from the database.
- An `UpdateBookService` class in `app/services/books_services/update_book_service.py` that updates a book's properties in the database.
- A `DeleteBookService` class in `app/services/books_services/delete_book_service.py` that deletes a book from the database.
- Corresponding tests for each service in the `tests/tests_book_services` directory.

These services add functionality to the API to retrieve, list, update, and delete books. 

## Testing

To test the changes, follow these steps:

1. Clone this repository and navigate to the root directory.
2. Install the required dependencies by running `poetry install`.
3. Start the Gremlin server by running `docker-compose up -d`.
4. Create a new virtual environment and activate it.
5. Start the API by running `uvicorn app.main:app`.
6. Use a tool like `curl`, `httpie`, or `Postman` to make requests to the API endpoints.
7. Verify that the responses match the expected results.
8. Run tests by executing the command `pytest` in the `tests` directory.
9. Verify that all tests pass.
